WT-4888 test/format data mismatch with BDB

keithbostic opened the pull request:
@agorrod, this branch removes Berkeley DB from the two places it's used in the source tree.

agorrod commented:
@sulabhM and @kommiharibabu I've assigned this review to you. @sulabhM it would be great if you can work through this with Hari - it touches the autoconf code, so it might be a good introduction to where that code lives.

keithbostic commented:
Thanks!

> dist/s_string.ok:bdb
> dist/s_string.ok:kvsbdb
> src/docs/spell.ok:bdb

Removed.

> .hgignore:^test/bt/(CONFIG|__rand|__wt.bdb|__wt.run|__wt.wt|db|t|vgout..*)

This is a Mercurial infrastructure file. It's not useful, I've removed it from the tree.

> test/format/util.c:	len = strlen(g.home) + strlen("bdb") + 2;
> test/format/util.c:	g.home_bdb = dmalloc(len);
> test/format/util.c:	testutil_check(__wt_snprintf(g.home_bdb, len, "%s/%s", g.home, "bdb"));
> test/format/format.h:	char home_bdb;	/ BDB directory */

Removed.

Pushed in 4432548.

keithbostic commented on test/format/README:
```diff
@@ -1,11 +0,0 @@
-The test/format program randomly generates WiredTiger databases with
```
There were only two paragraphs in the `README` and both of them were wrong, so there wasn't much to work with, I left it removed.

If you feel strongly that `format` needs a `README`, let me know and I'll write up something.
